src/core/id-headers.js

```diff
@@ -3,13 +3,32 @@
 // This is currently in core though it comes from a W3C rule. It may move in the future.
 
 export const name = "core/id-headers";
+import { addId } from "core/utils";
 
-export function run() {
+export function run(conf) {
   document
     .querySelectorAll(
-      "h2:not([id]), h3:not([id]), h4:not([id]), h5:not([id]), h6:not([id])"
+      `
+      section:not(.introductory) h2,
+      section:not(.introductory) h3,
+      section:not(.introductory) h4,
+      section:not(.introductory) h5,
+      section:not(.introductory) h6`
     )
     .forEach(elem => {
-      $(elem).makeID();
+      addId(elem);
+      if (conf.addSectionLinks) {
+        addSectionLink(elem);
+      }
     });
 }
+
+function addSectionLink(h) {
+  const section = h.closest("section[id]");
+  const a = document.createElement("a");
+  a.classList.add("self-link");
+  const id = section ? section.id : h.id;
+  a.href = `#${id}`;
+  a.setAttribute("aria-label", "§");
+  h.insertAdjacentElement("beforeend", a);
```

Comment:
Why not `h.apppendChild(a)`?

Reply:
``` js
function addSectionLink(h) {
  const section = h.closest("section[id]");
  const id = section ? section.id : h.id;
  h.appendChild(hyperHTML`
    <a href="${'#' + id}" class="self-link" aria-label="§"></a>
  `);
}
```

Reply:
I want it at the end tho :)

Reply:
oh wait, that is at end... I should stop